bin/vernemq.sh

```diff
@@ -157,8 +159,26 @@ sigterm_handler() {
         else
             terminating_node_name=VerneMQ@$IP_ADDRESS
         fi
-        /vernemq/bin/vmq-admin cluster leave node=$terminating_node_name -k > /dev/null
-        /vernemq/bin/vmq-admin node stop > /dev/null
+        kube_pod_names=$(curl -X GET $insecure --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt https://kubernetes.default.svc.$DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME/api/v1/namespaces/$DOCKER_VERNEMQ_KUBERNETES_NAMESPACE/pods?labelSelector=$DOCKER_VERNEMQ_KUBERNETES_LABEL_SELECTOR -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" | jq '.items[].spec.hostname' | sed 's/"//g' | tr '\n' ' ')
+        if [ $kube_pod_names == $MY_POD_NAME ]; then
+            echo "I'm the only pod remaining, not performing leave and state purge."
+            /vernemq/bin/vmq-admin node stop >/dev/null
+        else
+
+            statefulset=$(curl -X GET --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+                https://kubernetes.default.svc.$DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME/api/v1/namespaces/$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)/pods/$(hostname) | jq -r '.metadata.ownerReferences[0].name')
+
+            reschedule=$(curl -X GET --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+                https://kubernetes.default.svc.$DOCKER_VERNEMQ_KUBERNETES_CLUSTER_NAME/apis/apps/v1/namespaces/$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)/statefulsets/${statefulset} | jq '.status.replicas == .status.currentReplicas')
```

Comment:
Today we stumbled upon an issue with the current script and the helm chart it exist which is due to the service account not having the appropriate permissions to `get` `statefulsets` from `apps`.
This means we need to add the following rule
```
  - apiGroups: ["apps"]
    resources: ["statefulsets"]
    verbs: ["get"]
``` 
to the [role.yaml template](https://github.com/vernemq/docker-vernemq/blob/master/helm/vernemq/templates/role.yaml)
@isavcic can you modify the helm template to this PR just to help anyone that needs to test it.